package NYT.NJobTrackerClient.NProto;

import "yt/ytlib/misc/guid.proto";
import "yt/ytlib/misc/error.proto";
import "yt/ytlib/node_tracker_client/node.proto";

////////////////////////////////////////////////////////////////////////////////

message TJobInfo
{
    required NYT.NProto.TGuid job_id = 1;
    required int32 state = 2;
    optional NYT.NProto.TError error = 3;
}

message TJobStartInfo
{
    required int32 type = 1;
    required NYT.NProto.TGuid job_id = 2;
    required NYT.NProto.TGuid chunk_id = 3;
    repeated NYT.NNodeTrackerClient.NProto.TNodeDescriptor targets = 4;
}

message TJobStopInfo
{
    required NYT.NProto.TGuid job_id = 1;
}

////////////////////////////////////////////////////////////////////////////////

message TReqHeartbeat
{
    required int32 node_id = 1;
    repeated TJobInfo jobs = 2;
}

message TRspHeartbeat
{
    repeated TJobStartInfo jobs_to_start = 1;
    repeated TJobStopInfo jobs_to_stop = 2;
}

////////////////////////////////////////////////////////////////////////////////
